examples/1-advanced/40-mole_api_and_numba_jit.py

```diff
@@ -17,114 +17,124 @@
 
     aij = ai + aj
     Rab = Ra - Rb
     Rp = (ai * Ra + aj * Rb) / aij
     theta_ij = ai * aj / aij
-    scale = 1./np.sqrt(aij)
-    norm_fac *= scale**3 * np.exp(-theta_ij * Rab.dot(Rab))
-    x = roots * scale + Rp[:,None]
-    xa = x - Ra[:,None]
-    xb = x - Rb[:,None]
-
-    # Build mu = xa ** np.arange(li+1)[:,None,None]
-    # Build nu = xb ** np.arange(lj+1)[:,None,None]
+    scale = 1.0 / np.sqrt(aij)
+    norm_fac *= scale**3 * np.exp(-theta_ij * (Rab @ Rab))
+
     nroots = len(weights)
-    mu = np.empty((li+1,3,nroots))
-    nu = np.empty((lj+1,3,nroots))
-    for n in range(nroots):
-        powx = 1.
-        powy = 1.
-        powz = 1.
-        mu[0,0,n] = 1.
-        mu[0,1,n] = 1.
-        mu[0,2,n] = 1.
-        for i in range(1, li+1):
-            powx = powx * xa[0,n]
-            powy = powy * xa[1,n]
-            powz = powz * xa[2,n]
-            mu[i,0,n] = powx
-            mu[i,1,n] = powy
-            mu[i,2,n] = powz
-
-        powx = 1.
-        powy = 1.
-        powz = 1.
-        nu[0,0,n] = 1.
-        nu[0,1,n] = 1.
-        nu[0,2,n] = 1.
-        for i in range(1, lj+1):
-            powx = powx * xb[0,n]
-            powy = powy * xb[1,n]
-            powz = powz * xb[2,n]
-            nu[i,0,n] = powx
-            nu[i,1,n] = powy
-            nu[i,2,n] = powz
-
-    nfi = (li+1)*(li+2)//2
-    nfj = (lj+1)*(lj+2)//2
+    x = roots * scale + Rp[:, None]
+    xa = x - Ra[:, None]
+    xb = x - Rb[:, None]
+
+    mu = np.empty((li + 1, 3, nroots))
+    nu = np.empty((lj + 1, 3, nroots))
+    mu[0, :, :] = 1.0
+    nu[0, :, :] = 1.0
+
+    for d in range(3):
+        for p in range(1, li + 1):
+            mu[p, d, :] = mu[p - 1, d, :] * xa[d, :]
+        for p in range(1, lj + 1):
+            nu[p, d, :] = nu[p - 1, d, :] * xb[d, :]
+
+    nfi = (li + 1) * (li + 2) // 2
+    nfj = (lj + 1) * (lj + 2) // 2
     s = np.empty((nfi, nfj))
+
     i = 0
     for ix in range(li, -1, -1):
-        for iy in range(li-ix, -1, -1):
+        for iy in range(li - ix, -1, -1):
             iz = li - ix - iy
             j = 0
             for jx in range(lj, -1, -1):
-                for jy in range(lj-jx, -1, -1):
+                for jy in range(lj - jx, -1, -1):
                     jz = lj - jx - jy
-                    Ix = 0
-                    Iy = 0
-                    Iz = 0
+
+                    Ix = 0.0
+                    Iy = 0.0
+                    Iz = 0.0
                     for n in range(nroots):
-                        Ix += mu[ix,0,n] * nu[jx,0,n] * weights[n]
-                        Iy += mu[iy,1,n] * nu[jy,1,n] * weights[n]
-                        Iz += mu[iz,2,n] * nu[jz,2,n] * weights[n]
-                    s[i,j] = Ix * Iy * Iz * norm_fac
+                        w = weights[n]
+                        Ix += abs(mu[ix, 0, n] * nu[jx, 0, n] * w)
+                        Iy += abs(mu[iy, 1, n] * nu[jy, 1, n] * w)
+                        Iz += abs(mu[iz, 2, n] * nu[jz, 2, n] * w)
```

Comment:
You are assuming that the weight is positive here, which is not the case for some of the Lebedev rules. This code will give the wrong result.